### ISSUE TITLE
UI tests fail after restart of all pods

### ISSUE THREAD

**Issue by project-administrator** [trustbloc/sandbox#1081]
A restart of the pods triggers all UI tests to fail.

The issue can be reproduced in the local sandbox environment.

Steps to reproduce:
```
sandbox$ CPUS=6 MEMORY=8g make setup-deploy && sleep 1000 && make automation-test # start the local env and run the tests
## tests succeed

sandbox$ for pod in `kubectl get --no-headers=true pods -o name | grep -Ev 'mysql|couchdb|ipfs'`; do kubectl delete $pod; done # restart all pods except of the DBs
```
wait for the pods to restart and get into the Running state, and re-run the tests:
```
sandbox$ make automation-test
## tests fail
```
all tests will fail. In order to work around this issue, we can re-deploy the jobs:
```
sandbox/k8s/jobs$ make undeploy
sandbox/k8s/jobs$ make deploy
```
running the tests will succeed now: 
```
sandbox$ make automation-test
## tests succeed
```

Expected result: tests succeed after re-starting all of the pods *without* running (re-deploying) the jobs.

**Comment by alialkhalidi** [comment]
updating the restart step to reproduce the issue to include strapi/oathkeeper and list of current persistance-backends:

`
for pod in $(kubectl get --no-headers=true pods -o name | grep -Ev 'mq|pod/oathkeeper|strapi|postgres|mongodb|ipfs'); do kubectl delete $pod --wait=false; done
`

This will make all the test except new-bank-account.js test pass.
adapter-issuer logs:

`
[edge-adapter/issuerops] 2021/11/17 22:19:43 UTC - http.WriteErrorResponseWithLog -> ERROR endpoint=[/oidc/request] status=[500] errMsg=[server error]
`

re-running the issuer_adapter_configure.sh job script makes all the ui-tests to pass, and has the following results:

`
./issuer_adapter_configure.sh 
Adding Issuer Adapter profiles
'created' field from profile tb-cc-issuer response is: {"errMessage":"failed to create profile: profile tb-cc-issuer already exists"}
'created' field from profile tb-cr-issuer response is: {"errMessage":"failed to create profile: profile tb-cr-issuer already exists"}
'created' field from profile tb-dl-issuer response is: {"errMessage":"failed to create profile: profile tb-dl-issuer already exists"}
Finished adding Issuer Adapter profiles
`

@rolsonquadras @sudeshrshetty is this something we can fix in the adapter-issuer?

**Comment by alialkhalidi** [comment]
With the previous PR the remaining set of pods (cms/oathkeeper) are covered.
one followup item is to make the cms use profiles job idempotent; it should not fail when data (admin, user, profiles, etc) is already present.

**Comment by rolsonquadras** [comment]
> re-running the issuer_adapter_configure.sh job script makes all the ui-tests to pass, and has the following results:
> 
> `./issuer_adapter_configure.sh Adding Issuer Adapter profiles 'created' field from profile tb-cc-issuer response is: {"errMessage":"failed to create profile: profile tb-cc-issuer already exists"} 'created' field from profile tb-cr-issuer response is: {"errMessage":"failed to create profile: profile tb-cr-issuer already exists"} 'created' field from profile tb-dl-issuer response is: {"errMessage":"failed to create profile: profile tb-dl-issuer already exists"} Finished adding Issuer Adapter profiles`
> 
> @rolsonquadras @sudeshrshetty is this something we can fix in the adapter-issuer?

@alialkhalidi we shouldn't create a new profile on restarts - these are one time setups. Is there a way to separate these scripts from restart logic ?

**Comment by alialkhalidi** [comment]
I'll try to clarify... its not about re-running the jobs; in fact, that is not possible unless we re-deploy the jobs.
re-runing the script (from the job) make issuer-adapter work again rendering the ui-tests to the pass, without it, and with a simple restart, it does not.... something is not getting persisted across restarts in adapter-issuer, though the responses from the script state whats expected... data is there. if the profiles are there, why issuer-adapter needs these calls to be back?

**Comment by alialkhalidi** [comment]
with the latest sandbox/k8s, adapter logging added, and after a restart, adapter-issuer is showing these logs:

` [edge-adapter/issuerops] 2021/12/02 23:30:30 UTC - http.WriteErrorResponseWithLog -> ERROR endpoint=[/oidc/request] status=[500] errMsg=[get oidc client: constructing oidc client: failed to init oidc provider with url [tb-cr-issuer] : Get "tb-cr-issuer/.well-known/openid-configuration": unsupported protocol scheme ""]`

`
 [edge-adapter/issuerops] 2021/12/02 23:31:34 UTC - http.WriteErrorResponseWithLog -> ERROR endpoint=[/oidc/request] status=[500] errMsg=[get oidc client: constructing oidc client: failed to init oidc provider with url [tb-dl-issuer] : Get "tb-dl-issuer/.well-known/openid-configuration": unsupported protocol scheme ""]
`
I checked mongodb tables, the records show oidc pointing to hydra. the log message shows its missing https://hydra..... is the path correct though?

**Comment by rolsonquadras** [comment]
The issue has been fixes as part of https://github.com/trustbloc/adapter/issues/552.

@alialkhalidi can you please revalidate the scenario ?

**Comment by rolsonquadras** [comment]
@alialkhalidi would it be possible to share the issuer-adapter failure logs ?

**Comment by alialkhalidi** [comment]
issuer-adapter logs:
```
 [edge-adapter/wallet-bridge] 2021/12/16 14:50:12 UTC - http.WriteErrorResponseWithLog -> ERROR endpoint=[/wallet-bridge/get-preferences/{id}] status=[500] errMsg=[data not found]
 [edge-adapter/issuerops] 2021/12/16 14:50:17 UTC - http.WriteResponseWithLog -> INFO endpoint=[/issuer/didcomm/chapi/request] msg=[success]
 [edge-adapter/wallet-bridge] 2021/12/16 14:50:17 UTC - operation.(*Operation).SaveWalletPreferences -> INFO endpoint=[/wallet-bridge/save-preferences] msg=[success]
 [edge-adapter/wallet-bridge] 2021/12/16 14:50:25 UTC - operation.(*Operation).stateMsgListener -> WARNING Failed to update wallet application profile with invitationID=954db1b0-42ec-4b9b-861c-93d9d2472e51 and connectionID=c2401236-2d69-427b-ada3-f72075480596: failed to get user info for given invitation ID [954db1b0-42ec-4b9b-861c-93d9d2472e51]: data not found
 [aries-framework/out-of-band/service] 2021/12/16 14:50:25 UTC - n/a -> ERROR command=[out-of-band] action=[handleDIDEvent] [] errMsg=[handleDIDEvent: ParentThreadID is empty]
 [edge-adapter/msgsvc] 2021/12/16 14:50:25 UTC - route.(*Service).didCommMsgListener -> INFO msgType=[https://trustbloc.dev/blinded-routing/1.0/diddoc-req] id=[0474514b-28af-497a-b52b-26bc70ec750d] msg=[success]
 [edge-adapter/msgsvc] 2021/12/16 14:50:28 UTC - route.(*Service).didCommMsgListener -> INFO msgType=[https://trustbloc.dev/blinded-routing/1.0/register-route-req] id=[e43fdef8-9318-4f4c-b887-934a9f6a7f68] msg=[success]
 [edge-adapter/issuerops] 2021/12/16 14:50:29 UTC - http.WriteResponseWithLog -> INFO endpoint=[/connect/validate] msg=[success]
 [edge-adapter/issuerops] 2021/12/16 14:50:37 UTC - http.WriteErrorResponseWithLog -> ERROR endpoint=[/issuer/didcomm/chapi/request] status=[500] errMsg=[error creating reference credential : create credential : failed to execute http request: http request: 500 no valid credential scope]
```

**Comment by rolsonquadras** [comment]
The issue has been fixed in trustbloc/adapter#576.

Verified the changes locally:
1. deploy components
2. run automation tests - all pass 
3. restart pods `for pod in $(kubectl get --no-headers=true pods -o name | grep -Ev 'mq|local-kms|postgres|mongodb|ipfs'); do kubectl delete $pod --wait=false; done`
4. run automation test - all pass 

@alialkhalidi can you please verify this as well ?

**Comment by alialkhalidi** [comment]
@rolsonquadras verified and all ui-test pass now. Thanks!!.
closing.
